### PR TITLE
Deprecate app store in favor of derived store

### PR DESCRIPTION
**Pull request by djbarnwal**
## Checklist
- [x] Manual verification
- [ ] Unit test coverage
- [ ] E2E test coverage
- [x] Needs manual QA?

#### Issue addressed: 
Uses `page` store from sveltekit to derive active entity
Adds missing navigation events

### CODE REVIEW

File: web-common/src/layout/app-store.ts [hunk 1]
```diff
@@ -14,54 +17,86 @@ export interface ActiveEntity {
...
+        type: MetricsEventScreenName.Source,
+      };
+      break;
+    case "/(application)/model/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Model,
+      };
+      break;
+    case "/(application)/dashboard/[name]":
```
Comment (AdityaHegde) [inline web-common/src/layout/app-store.ts:46]: Add cloud routes as well.

File: web-common/src/features/models/workspace/inspector/EmbeddedSourceReference.svelte [hunk 2]
```diff
@@ -25,17 +32,31 @@
...
+      MetricsEventScreenName.Model,
+      MetricsEventScreenName.Source
+    );
+  }
 </script>
 
 <WithModelResultTooltip {modelHasError}>
-  <a
-    href="/source/{entry.name}"
+  <div
```
Comment (AdityaHegde) [inline web-common/src/features/models/workspace/inspector/EmbeddedSourceReference.svelte:49]: Perhaps leave the a tag with href as is. Just add the firing event handler.

File: web-common/src/layout/app-store.ts [hunk 3]
```diff
@@ -14,54 +17,86 @@ export interface ActiveEntity {
...
+  }
+
+  appStore.setActiveEntity(
+    activeEntity.name,
+    ScreenToEntityMap[activeEntity.type]
+  );
+  return activeEntity;
+});
+
+// TODO: Do we still need this?
```
Comment (AdityaHegde) [inline web-common/src/layout/app-store.ts:76]: You can add the previous entity name to appScreen and get rid of this. Call `httpRequestQueue.inactiveByName` directly in the body of `appScreen`.

**Review by AdityaHegde — CHANGES_REQUESTED** [review]

File: web-common/src/layout/app-store.ts (hunk 3 above)
Comment (djbarnwal) [reply web-common/src/layout/app-store.ts]: I want to keep `appScreen` clean as it's being heavily utilized across the application for telemetry. Better it just contains the current active entity. Will leave this as is for now.

File: web-common/src/layout/app-store.ts [hunk 4]
```diff
@@ -14,54 +17,91 @@ export interface ActiveEntity {
...
+// App store is being utilized for making previous entity inactive in the HTTP request queue
+const { update } = writable({
   activeEntity: undefined,
   previousActiveEntity: undefined,
 } as AppStore);
 
 const appStoreReducers = {
   setActiveEntity(name: string, type: EntityType) {
     update((state) => {
+      state.previousActiveEntity = state.activeEntity;
```
Comment (AdityaHegde) [inline web-common/src/layout/app-store.ts:91]: Why is this moved to the top? Wouldnt this make the current entity inactive? Either move this to the bottom or move the call to `setActiveEntity` to the top in `appScreen` body.

**Review by AdityaHegde — APPROVED** [review]
Just address the small comment. Good otherwise.

File: web-common/src/layout/app-store.ts (hunk 4 above)
Comment (djbarnwal) [reply web-common/src/layout/app-store.ts:91]: The previous implementation had a bug. Corrected it with these changes. You can verify by console logging `state.previousActiveEntity.name` when it's being made inactive. 

Now when you move from asset A to asset B, asset A is being made inactive now.

`name` provided through props is the current active entity being set. `state.activeEntity` just before being updated is now the previous active entity which has to be made inactive.
